fvtest/porttest/si.cpp

```diff
@@ -796,6 +796,106 @@ TEST(PortSysinfoTest, sysinfo_test_sysinfo_set_limit_CORE_FILE)
 	reportTestExit(OMRPORTLIB, testName);
 }
 
+	/**
+	 *
+	 * Test omrsysinfo_test_sysinfo_set_limit and omrsysinfo_test_sysinfo_get_limit
+	 * with resourceID OMRPORT_RESOURCE_FILE_DESCRIPTORS
+	 *
+	 */
+	TEST(PortSysinfoTest, sysinfo_test_sysinfo_set_limit_FILE_DESCRIPTORS)
+	{
+		OMRPORT_ACCESS_FROM_OMRPORT(portTestEnv->getPortLibrary());
+		const char *testName = "omrsysinfo_test_sysinfo_set_limit_FILE_DESCRIPTORS";
+		intptr_t rc = -1;
+		uint64_t originalCurLimit = 0;
+		uint64_t originalMaxLimit = 0;
+		uint64_t currentLimit = 0;
+		const uint64_t descriptorLimit = 256;
+
+		reportTestEntry(OMRPORTLIB, testName);
+
+		/* save original soft limit */
+		omrsysinfo_get_limit(OMRPORT_RESOURCE_FILE_DESCRIPTORS, &originalCurLimit);
+
+		rc = omrsysinfo_set_limit(OMRPORT_RESOURCE_FILE_DESCRIPTORS, descriptorLimit);
+		if (0 != rc) {
+			outputErrorMessage(PORTTEST_ERROR_ARGS, "omrsysinfo_set_limit FAILED\n");
+			reportTestExit(OMRPORTLIB, testName);
+			return;
+		}
+
+		omrsysinfo_get_limit(OMRPORT_RESOURCE_FILE_DESCRIPTORS, &currentLimit);
+		if (descriptorLimit == currentLimit) {
+			portTestEnv->log("omrsysinfo_set_limit set FILE_DESCRIPTORS soft max successful\n");
+		} else {
+			outputErrorMessage(PORTTEST_ERROR_ARGS, "omrsysinfo_set_limit set FILE_DESCRIPTORS soft max FAILED\n");
```

Comment:
Including the original limit, the new soft limit, and the result of get_limit in all the error messages would make this much easier to debug when it fails

Reply:
Fixed.